DependencyInjection/Compiler/AddProcessorsPass.php

```diff
@@ -28,32 +28,57 @@ public function process(ContainerBuilder $container)
             return;
         }
 
+        $processors = [];
+
         foreach ($container->findTaggedServiceIds('monolog.processor') as $id => $tags) {
             foreach ($tags as $tag) {
-                if (!empty($tag['channel']) && !empty($tag['handler'])) {
-                    throw new \InvalidArgumentException(sprintf('you cannot specify both the "handler" and "channel" attributes for the "monolog.processor" tag on service "%s"', $id));
+                if (!isset($tag['priority'])) {
+                    $tag['priority'] = 0;
                 }
 
-                if (!empty($tag['handler'])) {
-                    $definition = $container->findDefinition(sprintf('monolog.handler.%s', $tag['handler']));
-                } elseif (!empty($tag['channel'])) {
-                    if ('app' === $tag['channel']) {
-                        $definition = $container->getDefinition('monolog.logger');
-                    } else {
-                        $definition = $container->getDefinition(sprintf('monolog.logger.%s', $tag['channel']));
-                    }
-                } else {
-                    $definition = $container->getDefinition('monolog.logger_prototype');
-                }
+                $processors[] = [
+                    'id' => $id,
+                    'tag' => $tag,
+                ];
+            }
+        }
+
+        // Sort by priority so that higher-prio processors are added last.
+        // The effect is that monolog will call the higher-prio processors first
+        usort(
+            $processors,
+            function (array $left, array $right) {
+               return $left['tag']['priority'] - $right['tag']['priority'];
+            }
+        );
+
+        foreach ($processors as $processor) {
+            $tag = $processor['tag'];
+            $id = $processor['id'];
+
```

Comment:
Below this line the code is identical to what it was, except 1 level less indented